lib/init.c

```diff
@@ -270,7 +270,7 @@ void smb2_destroy_context(struct smb2_context *smb2)
         }
 
         if (smb2->fd != -1) {
-                close(smb2->fd);
+                fclose(smb2->fd);
```

Comment:
These changes look wrong.   smb2->fd is a file descriptor, not a FILE * structure so this should not even compile.